src/suites/cts/validation/vertex_input.spec.ts

```diff
@@ -0,0 +1,619 @@
+          ],
+        },
+        {
+          stride: 0,
+          attributeSet: [
+            {
+              shaderLocation: 0,
+              format: 'float',
+            },
+          ],
+        },
+      ],
+    };
+    // Test same shader location in two attributes in different buffers
+    const descriptor = t.getDescriptor(vertexInput, VERTEX_SHADER_CODE_WITH_NO_INPUT);
+
+    await t.expectValidationError(() => {
+      t.device.createRenderPipeline(descriptor);
+    });
+  }
+});
+
+g.test('check out of bounds condition on attribute shader location', async t => {
+  const vertexInput: GPUVertexInputDescriptor = {
+    vertexBuffers: [
+      {
+        stride: 0,
+        attributeSet: [
+          {
+            shaderLocation: MAX_VERTEX_ATTRIBUTES - 1,
+            format: 'float',
+          },
+        ],
+      },
+    ],
+  };
+  {
+    // Control case, setting last attribute shader location
+    const descriptor = t.getDescriptor(vertexInput, VERTEX_SHADER_CODE_WITH_NO_INPUT);
+    t.device.createRenderPipeline(descriptor);
+  }
+  {
+    // Test attribute location OOB
+    vertexInput.vertexBuffers![0].attributeSet![0].shaderLocation = MAX_VERTEX_ATTRIBUTES;
+    const descriptor = t.getDescriptor(vertexInput, VERTEX_SHADER_CODE_WITH_NO_INPUT);
+
+    await t.expectValidationError(() => {
+      t.device.createRenderPipeline(descriptor);
+    });
+  }
+});
+
+g.test('check attribute offset out of bounds', async t => {
+  const vertexInput: GPUVertexInputDescriptor = {
+    vertexBuffers: [
+      {
+        stride: 0,
+        attributeSet: [
+          {
+            offset: MAX_VERTEX_BUFFER_END - Float32Array.BYTES_PER_ELEMENT,
+            shaderLocation: 0,
+            format: 'float',
+          },
+        ],
+      },
+    ],
+  };
+  {
+    // Control case, setting max attribute offset for Float32 vertex format
+    const descriptor = t.getDescriptor(vertexInput, VERTEX_SHADER_CODE_WITH_NO_INPUT);
+    t.device.createRenderPipeline(descriptor);
+  }
+  {
+    // Test attribute offset out of bounds
+    vertexInput.vertexBuffers![0].attributeSet![0].offset = MAX_VERTEX_BUFFER_END - 1;
+    const descriptor = t.getDescriptor(vertexInput, VERTEX_SHADER_CODE_WITH_NO_INPUT);
+
+    await t.expectValidationError(() => {
+      t.device.createRenderPipeline(descriptor);
+    });
+  }
+});
+
+g.test('check attribute offset overflow', async t => {
+  const vertexInput: GPUVertexInputDescriptor = {
+    vertexBuffers: [
+      {
+        stride: 0,
+        attributeSet: [
+          {
+            offset: Number.MAX_SAFE_INTEGER,
+            shaderLocation: 0,
+            format: 'float',
+          },
+        ],
+      },
+    ],
+  };
+  const descriptor = t.getDescriptor(vertexInput, VERTEX_SHADER_CODE_WITH_NO_INPUT);
+
+  await t.expectValidationError(() => {
+    t.device.createRenderPipeline(descriptor);
+  });
+});
+
+g.test('check for some potential underflow in the vertex input validation', async t => {
```

Comment:
Can you clarify what this is trying to test? It looks like it's making an attribute (size 16) that's larger than the stride (4).

Reply:
According to https://dawn-review.googlesource.com/c/dawn/+/7840,  it's trying to test that  no underflow is possible because the max vertex format size is smaller than 2048. @Kangz may know better

Reply:
After offline discussion with @Kangz, we decided to remove this test as it was not needed in the CTS.